src/middleware/traceUser.ts

```diff
@@ -0,0 +1,36 @@
+import { NextFunction, Request, Response } from "express"
+import { Session } from "express-session"
+
+import tracer from "@utils/tracer"
+
+import { SessionData } from "@root/types/express/session"
+
+type UserSession = Session & SessionData
+
+function isUserSession(session: Session): session is UserSession {
+  return !!(
+    session &&
+    "id" in session &&
+    "userInfo" in session &&
+    typeof session.userInfo === "object" &&
+    session.userInfo &&
+    "isomerUserId" in session.userInfo
+  )
+}
```

Comment:
Typescript is a pain in my ass 😢 

It feels way overkill to do this, but I couldn't get the types to work otherwise 😢

Reply:
hmm wait this is not a strict validation tho right? should we just add the remaining fields? Feels like this is more risky than just doing a type assertion of just `req: Request<UserSession>` since this in first impression feels like 'correct and internal type checking'